source/fuzz/fuzzer_pass_add_loads.cpp

```diff
@@ -79,13 +83,46 @@ void FuzzerPassAddLoads::Apply() {
           return;
         }
 
-        // Choose a pointer at random, and create and apply a loading
-        // transformation based on it.
-        ApplyTransformation(TransformationLoad(
-            GetFuzzerContext()->GetFreshId(),
+        auto chosen_instruction =
             relevant_instructions[GetFuzzerContext()->RandomIndex(
-                                      relevant_instructions)]
-                ->result_id(),
+                relevant_instructions)];
+
+        bool is_atomic_load = false;
+        uint32_t memory_scope_id = 0;
+        uint32_t memory_semtantics_id = 0;
+
+        // has work group or storage buffer (storage class)
+        if (chosen_instruction->GetInOperand(0).words[0] ==
+                SpvMemorySemanticsUniformMemoryMask ||
+            chosen_instruction->GetInOperand(0).words[0] ==
+                SpvMemorySemanticsWorkgroupMemoryMask) {
+          if (GetFuzzerContext()->ChoosePercentage(
+                  GetFuzzerContext()->GetChanceOfAddingAtomicLoad())) {
+            is_atomic_load = true;
+
+            memory_scope_id = FindOrCreateConstant(
+                {SpvScopeInvocation},
+                FindOrCreateIntegerType(32, GetFuzzerContext()->ChooseEven()),
+                false);
+
+            memory_semtantics_id = FindOrCreateConstant(
+                {static_cast<uint32_t>(
+                    chosen_instruction->GetInOperand(0).words[0] ==
+                            SpvMemorySemanticsUniformMemoryMask
+                        ? SpvMemorySemanticsUniformMemoryMask
+                        : SpvMemorySemanticsUniformMemoryMask)},
+                FindOrCreateIntegerType(32, GetFuzzerContext()->ChooseEven()),
+                false);
+          }
+        }
+
+        // Choose a pointer at random, also choose a probabilistic chance for
+        // atomic load, if atomic load become true so will create constants for
+        // memory scope and semantics else they will be still zero, then create
+        // and apply a loading transformation based on a pointer and is_atomic.
```

Comment:
Sorry, that is not what I meant. I suggest describing what is happening below the comment. Delete text that describes what is happening above the comment. So I suggest you just change it to: create and apply the transformation.

Or just remove the comment entirely, in this case.

Reply:
Done.